dDatabase/CDClientDatabase/CDClientManager.cpp

```diff
@@ -52,43 +52,50 @@
-	CDZoneTableTable::Instance().LoadValuesFromDatabase();
+	if (CDClientDatabase::isConnected) {
+		CDActivityRewardsTable::Instance().LoadValuesFromDatabase();
+		CDActivitiesTable::Instance().LoadValuesFromDatabase();
+		CDCLIENT_DONT_CACHE_TABLE(CDAnimationsTable::Instance().LoadValuesFromDatabase());
+		CDBehaviorParameterTable::Instance().LoadValuesFromDatabase();
+		CDBehaviorTemplateTable::Instance().LoadValuesFromDatabase();
+		CDBrickIDTableTable::Instance().LoadValuesFromDatabase();
+		CDCLIENT_DONT_CACHE_TABLE(CDComponentsRegistryTable::Instance().LoadValuesFromDatabase());
+		CDCurrencyTableTable::Instance().LoadValuesFromDatabase();
+		CDDestructibleComponentTable::Instance().LoadValuesFromDatabase();
+		CDEmoteTableTable::Instance().LoadValuesFromDatabase();
+		CDFeatureGatingTable::Instance().LoadValuesFromDatabase();
+		CDInventoryComponentTable::Instance().LoadValuesFromDatabase();
+		CDCLIENT_DONT_CACHE_TABLE(CDItemComponentTable::Instance().LoadValuesFromDatabase());
+		CDItemSetSkillsTable::Instance().LoadValuesFromDatabase();
+		CDItemSetsTable::Instance().LoadValuesFromDatabase();
+		CDLevelProgressionLookupTable::Instance().LoadValuesFromDatabase();
+		CDCLIENT_DONT_CACHE_TABLE(CDLootMatrixTable::Instance().LoadValuesFromDatabase());
+		CDCLIENT_DONT_CACHE_TABLE(CDLootTableTable::Instance().LoadValuesFromDatabase());
+		CDMissionEmailTable::Instance().LoadValuesFromDatabase();
+		CDMissionNPCComponentTable::Instance().LoadValuesFromDatabase();
+		CDMissionTasksTable::Instance().LoadValuesFromDatabase();
+		CDMissionsTable::Instance().LoadValuesFromDatabase();
+		CDMovementAIComponentTable::Instance().LoadValuesFromDatabase();
+		CDObjectSkillsTable::Instance().LoadValuesFromDatabase();
+		CDCLIENT_DONT_CACHE_TABLE(CDObjectsTable::Instance().LoadValuesFromDatabase());
+		CDPhysicsComponentTable::Instance().LoadValuesFromDatabase();
+		CDPackageComponentTable::Instance().LoadValuesFromDatabase();
+		CDPetComponentTable::Instance().LoadValuesFromDatabase();
+		CDProximityMonitorComponentTable::Instance().LoadValuesFromDatabase();
+		CDPropertyEntranceComponentTable::Instance().LoadValuesFromDatabase();
+		CDPropertyTemplateTable::Instance().LoadValuesFromDatabase();
+		CDRailActivatorComponentTable::Instance().LoadValuesFromDatabase();
+		CDRarityTableTable::Instance().LoadValuesFromDatabase();
+		CDRebuildComponentTable::Instance().LoadValuesFromDatabase();
+		CDRewardCodesTable::Instance().LoadValuesFromDatabase();
+		CDRewardsTable::Instance().LoadValuesFromDatabase();
+		CDScriptComponentTable::Instance().LoadValuesFromDatabase();
+		CDSkillBehaviorTable::Instance().LoadValuesFromDatabase();
+		CDVendorComponentTable::Instance().LoadValuesFromDatabase();
+		CDZoneTableTable::Instance().LoadValuesFromDatabase();
+	} else {
+		LOG("Unable to connect to CDClientDatabase! Using default values instead!");
+
+		CDPetComponentTable::Instance().LoadValuesFromDefaults();
+	}
```

Comment:
Wouldn't it be catastrophic if the server failed to connect to the cdclient; feels like this should throw. It would regardless be more readable to invert the if here and get rid of the else clause with an early return.

Reply:
I believe it is like this for the testing, though I agree it does flow weird and doesn't make sense to continue execution if there is no connection.  Is there possibly a different way to communicate the same result jade?

Reply:
It is like this for the testing. I'll see if I can figure out a better way to handle this. Perhaps I can make some sort of "testing mode" flag and only have it load the fallbacks if that is set to true. It would be useful to have a mock CDClient database for our unit testing without actually connecting to it.

Reply:
The "standard" way to do this might be to have one constructor that takes a filename and loads the DB from SQLite and one without parameters that just creates an empty instance. Since these are all singletons that depend on ordering anyway, it may make sense to move the loading out of the Manager constructor and into the CDClientDatabase::Connect method or something that is called closer to it.

Right now, if someone were to call CDClientManager::Instance _before_ connect, things would just break (Null pointer dereference [here](https://github.com/DarkflameUniverse/DarkflameServer/blob/e4469f997ec10c63e9e1c3972542ae6b0cfdfe94/dDatabase/CDClientDatabase/CDClientDatabase.cpp#L14) would be my guess, which may leave the static in CDClientManager::Instance in a broken state even if we were to recover from the null ptr dereference.)

Reply:
I ended up doing something similar to that. Works a lot nicer than the old solution, IMO.